app/controllers/student_applications_controller.rb

```diff
@@ -0,0 +1,29 @@
+class StudentApplicationsController < ApplicationController
+  before_action :authenticate_applicant!
+  before_action :unique_semester_application
+
+  def new
+    @student_application = current_user.student_applications.build
+  end
+
+  def create
+    @student_application = current_user.student_applications.build student_application_params
+    @student_application.semester = Semester.current_semester
+    if @student_application.save
+      redirect_to students_apply_path, flash: { success: "You have"\
+      "successfully submitted an application. You will receive an email shortly." }
```

Comment:
Nitpick but could this line be indented just so it's clear that this string is not a new statement? And the one below too.